persistent-postgresql/Database/Persist/Postgresql.hs

```diff
@@ -700,6 +700,10 @@ getColumns getter def = do
     us <- with (stmtQuery stmt' vals) (\src -> runConduit $ src .| helperU)
     return $ cs ++ us
   where
+    refMap = Map.fromList $ foldl ref [] cols
+        where ref rs c = case cReference c of
+                  Nothing -> rs
+                  (Just r) -> (unDBName $ cName c, r) : rs
```

Comment:
Here we building a map of column names to foreign keys, so we can pass that on to `getColumn`.